sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedMicroBatchStream.scala

```diff
@@ -40,7 +40,7 @@ private class ChangeFeedMicroBatchStream
   private val changeFeedConfig = CosmosChangeFeedConfig.parseCosmosChangeFeedConfig(config)
   private val client = CosmosClientCache(clientConfiguration, Some(cosmosClientStateHandle))
   private val container = ThroughputControlHelper.getContainer(config, containerConfig, client)
-  container.openConnectionsAndInitCaches().block()
+  SparkUtils.safeOpenConnectionInitCaches(container, log)
```

Comment:
Is there any test we can add to avoid any future changes reverting this behavior?

Reply:
ideally this should go to the sdk not the spark connector. It is not worth to add a test for try/catch pattern.

also not possible to add future proof test for future code which is not written yet.